contracts/market-wrapper/NounsMarketWrapper.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports ============
+import {INounsAuctionHouse} from "../external/interfaces/INounsAuctionHouse.sol";
+
+// ============ Internal Imports ============
+import {IMarketWrapper} from "./IMarketWrapper.sol";
+
+/**
+ * @title NounsMarketWrapper
+ * @author Anna Carroll + Nounders
+ * @notice MarketWrapper contract implementing IMarketWrapper interface
+ * according to the logic of Nouns' Auction Houses
+ */
+contract NounsMarketWrapper is IMarketWrapper {
+    // ============ Public Immutables ============
+
+    INounsAuctionHouse public immutable market;
+
+    // ======== Constructor =========
+
+    constructor(address _nounsAuctionHouse) {
+        market = INounsAuctionHouse(_nounsAuctionHouse);
+    }
+
+    // ======== External Functions =========
+
+    /**
+     * @notice Determine whether there is an existing, active auction
+     * for this token. In the Nouns auction house, the current auction
+     * id is the token id, which increments sequentially, forever.
+     * @return TRUE if the auction exists
+     */
+    function auctionExists(uint256 auctionId)
+      public
+      view
+      override
+      returns (bool)
+    {
+        (uint256 currentAuctionId, , , , , ) = market.auction();
+        return auctionId == currentAuctionId;
```

Comment:
this function is meant to ensure that the auction in question is both existing & **active** 
if `auction.settled` is true, then this should return false.